### PR TITLE
Add Django 4 testing and remove unsupported Python and Django versions

### PR DESCRIPTION
**Pull request by vchrisb**
* removed unsupported Django Versions `2.2, 3.0, 3.1` and Python Version `3.6`
* added Django `4.0` and Python `3.10`
* Updated Mysql and Postgresql Versions
* Updated Github Actions
* fixed pre-commit.ci by adding basic pre-commit-config

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,12 +9,12 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.6', '3.7', '3.8', '3.9']
-        django-version: ['2.2', '3.0', '3.1', '3.2', 'main']
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+        django-version: ['3.2', '4.0']
```
Comment (clintonb) [inline .github/workflows/test.yml:13]: “Main” is missing.

File: tox.ini [hunk 2]
```diff
@@ -1,35 +1,31 @@
...
 [gh-actions:env]
 DJANGO =
-    2.2: dj22
-    3.0: dj30
-    3.1: dj31
-    3.2: dj32
-    main: djmain
+    3.2: django
+    4.0: django40
+    main: djangomain
```
Comment (clintonb) [inline tox.ini:20]: djmain

**Review by clintonb — CHANGES_REQUESTED** [review]
Fix the test config, and this should be all set.

**Review by clintonb — APPROVED** [review]

**Comment by vchrisb** [comment]
please don't merge yet. There is still a mistake...

File: tox.ini [hunk 3]
```diff
@@ -1,36 +1,32 @@
...
+  3.10: py310
 
 [gh-actions:env]
 DJANGO =
-    2.2: dj22
-    3.0: dj30
-    3.1: dj31
-    3.2: dj32
-    main: djmain
+    3.2: django
```
Comment (vchrisb) [inline tox.ini:18]: This should be django32

**Review by clintonb — CHANGES_REQUESTED** [review]
Awaiting change from engineer.

**Comment by clintonb** [comment]
@vchrisb can you update the classifiers in setup.py, too?

**Review by clintonb — APPROVED** [review]

**Comment by vchrisb** [comment]
I'll create another pr to increase the test coverage once this is merged.

**Pull request merged**